virtcontainers/kata_agent.go

```diff
@@ -1401,13 +1410,17 @@ func (k *kataAgent) handleBlockVolumes(c *Container) []*grpc.Storage {
 			k.Logger().Error("malformed block drive")
 			continue
 		}
-		if c.sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioBlock {
+		switch {
+		case c.sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioBlockCCW:
+			vol.Driver = kataBlkCCWDevType
+			vol.Source = blockDrive.DevNo
+		case c.sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioBlock:
 			vol.Driver = kataBlkDevType
 			vol.Source = blockDrive.PCIAddr
-		} else if c.sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioMmio {
+		case c.sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioMmio:
 			vol.Driver = kataMmioBlkDevType
 			vol.Source = blockDrive.VirtPath
-		} else {
+		case c.sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioSCSI:
```

Comment:
There should not be any unknown Drivers here

Reply:
But adding a `default` would make the test exhaustive and flag any possible issues in the future.

In fact, looking more closely, I think `handleBlockVolumes()` should return an `error` too as it's already (trying to) handle two error scenarios which it looks like the caller ignores (since it's not obvious this function can "fail".

Reply:
Added `error` to the return of `handleBlockVolumes()` and made the default for both a return with `error`.